src/cloud-api-adaptor/test/e2e/assessment_runner.go

```diff
@@ -416,6 +423,25 @@ func (tc *TestCase) Run() {
 					tc.assert.HasPodVM(t, tc.pod.Name)
 				}
 
+				if tc.podState != v1.PodRunning {
+					if tc.FailReason != "" {
+						jsonData, err := json.Marshal(tc.pod.Status)
+						if err != nil {
+							fmt.Println("Error marshaling pod.Status to JSON: ", err.Error())
+						} else {
+							t.Logf("Current Pod State: %v", string(jsonData))
+						}
+						if strings.Contains(string(jsonData), tc.FailReason) {
+							log.Infof("failed due to expected reason %s", tc.FailReason)
+						} else {
+							t.Fatal(err)
+						}
+					} else {
+						log.Infof("Pod Failed If you want to cross check please give .WithFailReason(error string)")
+					}
+					tc.assert.HasPodVM(t, tc.pod.Name)
```

Comment:
If pod is not running, the Pod VM may be not created in some case. Is it necessary to check Pod VM existence here?